app/processing/process-payment-request.js

```diff
@@ -18,6 +20,17 @@ const processPaymentRequest = async (scheduledPaymentRequest) => {
 
   const paymentRequests = await transformPaymentRequest(paymentRequest)
   const { deltaPaymentRequest, completedPaymentRequests } = paymentRequests
+
+  // if FRN is closed (SFI only), remove AR
+  if (paymentRequest.schemeId === SFI && config.handleSFIClosures) {
+    const closedFRN = await getClosedFRN(paymentRequest.frn)
+    if (closedFRN !== null) {
+      console.log(`FRN ${paymentRequest.frn} has been closed, skipping request editor and holds`)

```

Comment:
Suggest adding a scheme reference to this error so it's clear which payment for the customer is affected.  Maybe the agreement number too?

Also would remove the reference to holds.  True it's skipping the auto hold functionality for top ups and recoveries.  But the general hold logic would prevent the payment even being picked up for processing earlier in the service and we still want that to happen.  So might be misleading

Reply:
console log amended to include the relevant information